### DISCUSSION TITLE
Can USB serial be used normally

### DISCUSSION THREAD

**Discussion by liqijian101** [luc-github/ESP3D-TFT#26] [Q&A] [answered]
// NOTE:
  // this location allows usb-host driver to be installed - later it will failed
  // Do not know why...
  if (usb_serial_init() != ESP_OK) {
    return ESP_FAIL;
  }

**Comment by luc-github** [comment]
What do you mean ?

**Reply by liqijian101** [reply]
The following content has been annotated in the bsp. c text
// NOTE:
// this location allows usb-host driver to be installed - later it will failed
// Do not know why...

So may I ask ESP3D_ USB_ SERIAL_FEATURE  can be used normally

**Reply by luc-github** [reply]
yes on S3 only

**Reply by luc-github** [reply]
and parameters are in settings

**Reply by liqijian101** [reply]
When I start DESP3D_ USB_ SERIAL_ FEATURE, and  connecting the printer, ESP3D_ TFT always restarts

**Reply by luc-github** [reply]
what do you mean I start DESP3D_ USB_ SERIAL_ FEATURE ?
you do not have to touch anything in CMakelists.txt for that , it is already implemented on supported TFT https://github.com/luc-github/ESP3D-TFT/blob/main/CMakeLists.txt#L249, https://github.com/luc-github/ESP3D-TFT/blob/main/CMakeLists.txt#L273

**Reply by liqijian101** [reply]
I have created a new board
Therefore, I have added a configuration for ESP32S3 in CMakelists.txt, I have configured add accordingly_ Compile_ Options (- DEP3D-USB_SERIAL-FEATURE=1)
I have added a new board, ESP32S3_ BZM_ TFT35_ GT911
[CMakeLists.txt](https://github.com/luc-github/ESP3D-TFT/files/12447110/CMakeLists.txt)

**Reply by luc-github** [reply]
did you added and configured the USB serial drivers ?

**Reply by liqijian101** [reply]
Yes.
Maybe it is wrong with the board

**Reply by luc-github** [reply, marked as answer]
I have just enabled the USB serial in settings of  HMI V3 TFT and restarted board and I do not have reset - so I cannot say what is your issue - you need to enable the log and check why the TFT reset - it may help
